File: src/main/java/br/com/zumbolovsky/fateapp/service/TestAProcessorService.java
package br.com.zumbolovsky.fateapp.service;

import org.springframework.stereotype.Component;

@Component
public class TestAProcessorService implements TestProcessorService {

    @Override
    public String process() {
        return "I'm implementation A";
    }

    @Override
    public boolean supports(TestEnum delimiter) {
        return TestEnum.TEST_A == delimiter;
    }
}

File: src/main/kotlin/br/com/zumbolovsky/fateapp/TestAProcessorService.kt
package br.com.zumbolovsky.fateapp

import org.springframework.stereotype.Component

@Component
class TestAProcessorService : TestProcessorService {
    override fun process(): String {
        return "I'm implementation A"
    }

    override fun supports(delimiter: TestEnum): Boolean {
        return TestEnum.TEST_A == delimiter
    }
}